molten-core/src/main/java/com/hotels/molten/core/collapser/FanOutRequestCollapser.java

```diff
@@ -397,6 +419,19 @@ public Builder(Function<List<C>, Mono<List<V>>> bulkProvider) {
             return this;
         }
 
+        /**
+         * Sets the maximum time to wait for a prepared batch call if {@link #withBatchMaxConcurrency(int) max concurrency} is already fulfilled.
```

Comment:
This documentation is not exactly clear for me. How about:
`Sets the maximum time to wait for executing a prepared batch call if there are already {@link #withBatchMaxConcurrency(int) max concurrency} batches are running.`
Also add what happens if it times out.

Reply:
Thanks, added improvements in 335673c6. Let me know if it's good enough now or not!